File: testdata/jennies/builders/map_of_disjunctions/GoBuilder/map_of_disjunctions/librarypanel_builder_gen.go
package map_of_disjunctions

import (
	cog "github.com/grafana/cog/generated/cog"
)

var _ cog.Builder[LibraryPanel] = (*LibraryPanelBuilder)(nil)

type LibraryPanelBuilder struct {
    internal *LibraryPanel
    errors cog.BuildErrors
}

func NewLibraryPanelBuilder() *LibraryPanelBuilder {
	resource := NewLibraryPanel()
	builder := &LibraryPanelBuilder{
		internal: resource,
		errors: make(cog.BuildErrors, 0),
	}
    builder.internal.Kind = "Library"

	return builder
}



func (builder *LibraryPanelBuilder) Build() (LibraryPanel, error) {
	if err := builder.internal.Validate(); err != nil {
		return LibraryPanel{}, err
	}
	
	if len(builder.errors) > 0 {
	    return LibraryPanel{}, cog.MakeBuildErrors("map_of_disjunctions.libraryPanel", builder.errors)
	}

	return *builder.internal, nil
}

func (builder *LibraryPanelBuilder) Text(text string) *LibraryPanelBuilder {
    builder.internal.Text = text

    return builder
}

File: testdata/jennies/builders/map_of_disjunctions/TypescriptBuilder/src/mapOfDisjunctions/libraryPanelBuilder.gen.ts
import * as cog from '../cog';
import * as mapOfDisjunctions from '../mapOfDisjunctions';

export class LibraryPanelBuilder implements cog.Builder<mapOfDisjunctions.LibraryPanel> {
    protected readonly internal: mapOfDisjunctions.LibraryPanel;

    constructor() {
        this.internal = mapOfDisjunctions.defaultLibraryPanel();
        this.internal.kind = "Library";
    }

    /**
     * Builds the object.
     */
    build(): mapOfDisjunctions.LibraryPanel {
        return this.internal;
    }

    text(text: string): this {
        this.internal.text = text;
        return this;
    }
}
